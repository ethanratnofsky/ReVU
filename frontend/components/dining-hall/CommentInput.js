import { useState } from 'react';
import { Alert, Image, Keyboard, KeyboardAvoidingView, Pressable, TextInput, View } from 'react-native';

import commentInputStyles from './CommentInputStyles';

const CommentInput = ({ diningHallId }) => {
    const [comment, setComment] = useState('');

    const handleSubmit = () => {
        Alert.alert('Comment Posted', comment); // TODO: post comment to backend

        // TODO: only execute if comment was successfully posted
        setComment('');
        Keyboard.dismiss();
    };

    return (
        <KeyboardAvoidingView behavior='padding'>
            <View style={commentInputStyles.container}>
                <TextInput
                    style={commentInputStyles.commentInput}
                    placeholder={'Write a comment...'}
                    placeholderTextColor={'#999'}
                    value={comment}
                    onChangeText={setComment}
                />
                <Pressable onPress={handleSubmit} disabled={!comment}>
                    <Image style={commentInputStyles.send} source={comment ? require('../../assets/images/gold-send.png') : require('../../assets/images/gray-send.png')} />
                </Pressable>
            </View>
        </KeyboardAvoidingView>
    )
};

export default CommentInput;