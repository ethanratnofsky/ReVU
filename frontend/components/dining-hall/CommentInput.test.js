import React from 'react';
const renderer = require('react-test-renderer');
import CommentInput from './CommentInput';

describe('<CommentInput />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<CommentInput />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<CommentInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });