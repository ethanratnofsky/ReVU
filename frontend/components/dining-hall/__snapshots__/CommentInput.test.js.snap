// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#333",
        "borderRadius": 5,
        "color": "#FFFFFF",
        "flexDirection": "row",
        "marginHorizontal": 15,
        "marginVertical": 10,
        "width": 350,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Write a comment..."
      placeholderTextColor="#999"
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#FFFFFF",
          "flex": 1,
          "marginHorizontal": 10,
          "paddingVertical": 12,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginRight": 5,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
  </View>
</View>
`;
